nmostesting/mocks/Auth.py

```diff
@@ -0,0 +1,535 @@
+                ctx.check_hostname = False
+                ctx.verify_mode = ssl.CERT_NONE
+
+            # mock secondary Authorization server
+            auth_app = Flask(__name__)
+            auth_app.debug = False
+            auth_app.config["AUTH_INSTANCE"] = 1
+            self.auth.private_keys = [key_file]
+            port = self.auth.port
+            auth_app.register_blueprint(AUTH_API)
+
+            self.server = make_server("0.0.0.0", port, auth_app, threaded=True, ssl_context=ctx)
+            self.server_thread = threading.Thread(
+                target=self.server.serve_forever)
+            self.server_thread.daemon = True
+            self.server_thread.start()
+
+    def shutdown(self):
+        """Stop Authorization server"""
+        if self.server_thread:
+            self.server.shutdown()
+            self.server_thread.join()
+            self.server_thread = None
+
+
+class Auth(object):
+    def __init__(self, port_increment, version="v1.0"):
+        self.port = PORT_BASE + 9 + port_increment
+        self.private_keys = KEYS_MOCKS
+        self.version = version
+        self.protocol = "http"
+        self.host = get_default_ip()
+        if ENABLE_HTTPS:
+            self.protocol = "https"
+            self.host = get_mocks_hostname()
+        # authorization code of the authorization code flow
+        self.code = None
+
+    def make_mdns_info(self, priority=0, api_ver=None, ip=None):
+        """Get an mDNS ServiceInfo object in order to create an advertisement"""
+        if api_ver is None:
+            api_ver = self.version
+
+        api_proto = self.protocol
+
+        if ip is None:
+            ip = get_default_ip()
+            hostname = "nmos-mocks.local."
+        else:
+            hostname = ip.replace(".", "-") + ".local."
+
+        port = self.port
+
+        # TODO: Add another test which checks support for parsing CSV string in api_ver
+        txt = {'api_ver': api_ver, 'api_proto': api_proto, 'pri': str(priority)}
+
+        service_type = "_nmos-auth._tcp.local."
+        info = ServiceInfo(service_type,
+                           "NMOSTestSuite{}{}.{}".format(port, api_proto, service_type),
+                           addresses=[socket.inet_aton(ip)], port=port,
+                           properties=txt, server=hostname)
+        return info
+
+    def make_issuer(self):
+        return "{}://{}:{}".format(self.protocol, self.host, self.port)
+
+    def make_metadata(self):
+        metadata = {
+            "issuer": self.make_issuer(),
+            "authorization_endpoint": "{}://{}:{}/auth".format(self.protocol, self.host, self.port),
+            "token_endpoint": "{}://{}:{}/token".format(self.protocol, self.host, self.port),
+            "jwks_uri": "{}://{}:{}/jwks".format(self.protocol, self.host, self.port),
+            "registration_endpoint": "{}://{}:{}/register".format(self.protocol, self.host, self.port),
```

Comment:
Would it be better to follow the examples in the IS-10 spec (in which case `/authorize` rather than `/auth`), or perhaps to explicitly _**not**_ follow the examples (so maybe `/testauthorize`, etc.)?

Reply:
should use `/authorize`

Reply:
@jonathan-r-thorpe thoughts? wouldn't using `/testauthorize` etc. catch out implementations that hardwired the URL paths?

Reply:
Using the `/authorize` as it is matching to the IS-10 spec example. It is because all other endpoints used by the mock Auth server already match the spec examples.

We can use whatever value we want, as the final value is going to be provided by the authorization metadata's `authorization_endpoint`